Marlin/src/HAL/STM32/msc_sd.cpp

```diff
@@ -58,39 +62,41 @@ class Sd2CardUSBMscHandler : public USBMscHandler {
     // single block
     if (blkLen == 1) {
       hal.watchdog_refresh();
-      sd2card->writeBlock(blkAddr, pBuf);
-      return true;
+      return sd2card->writeBlock(blkAddr, pBuf);
     }
 
-    // multi block optimization
-    sd2card->writeStart(blkAddr, blkLen);
+    bool success = true;
+    sd2card->writeStart(blkAddr);
     while (blkLen--) {
       hal.watchdog_refresh();
-      sd2card->writeData(pBuf);
+      uint8_t multi_retry_cnt = SD_MULTIBLOCK_READ_RETRY_CNT;
+      while (!(success = sd2card->writeData(pBuf)) && multi_retry_cnt--) { /* retry... */ }
```

Comment:
My question with this concept is, does `sd2card->writeData` advance its block address on failure, or only on success? If it only advances it on failure, then this will work, otherwise I'll revert / rework it to hit `writeStop`/`writeStart` again before retry. But in that case, by advancing the `blkAddr` and still retrying the same block.

I think that makes sense, but mostly I hope it made sense that `sd2card` doesn't advance the buffer pointer until it succeeds.

Reply:
I think the restarting from beginning would make more sense because the retrying from the same region would be already implemented in the writeData method itself.